config/freebsd

```diff
@@ -20,110 +20,121 @@
-# `--version' to see if the compiler will print a version string.  You
-# can use the value of $FOO_BASENAME which is the base name of the
-# first word in $FOO, where FOO is either CC or F77 (note that the
-# value of CC may have changed above).
-
-case $CC_BASENAME in
-  gcc)
-    CFLAGS="$CFLAGS"
-    DEBUG_CFLAGS="-g -ansi -Wall -pedantic "
-    DEBUG_CPPFLAGS=
-    PROD_CFLAGS="-ansi -Wall -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Wmissing-prototypes -Wnested-externs -pedantic -O2"
-    PROD_CPPFLAGS=
-    PROFILE_CFLAGS="-pg"
-    PROFILE_CPPFLAGS=
-    ;;
-
-  *)
-    CFLAGS="$CFLAGS -ansi"
-    DEBUG_CFLAGS="-g"
-    DEBUG_CPPFLAGS=
-    PROD_CFLAGS="-O"
-    PROD_CPPFLAGS=
-    PROFILE_CFLAGS="-pg"
-    PROFILE_CPPFLAGS=
-    ;;
+# Figure out GNU FC compiler flags
+. $srcdir/config/gnu-fflags
+
+# compiler version strings
+
+# check if the compiler_version_info is already set
+if test -z "$cc_version_info"; then
+
+case $CC in
+    # whatever matches *pgcc* will also match *gcc*, so this one must come first
+    *pgcc*)
+        cc_version_info=`$CC $CFLAGS $H4_CFLAGS -V 2>&1 | grep 'pgcc'`
+        ;;
+
+    *gcc*)
+        cc_version_info=`$CC $CFLAGS $H4_CFLAGS --version 2>&1 | grep -v 'PathScale' |\
+            grep 'GCC' | sed 's/\(.*(GCC) [-a-z0-9\. ]*\).*/\1/'`
+        ;;
+
+    *icc*)
+        cc_version_info=`$CC $CCFLAGS $H4_CCFLAGS -V 2>&1 | grep 'Version' |\
+            sed 's/\(Intel.* Compiler\).*\( Version [a-z0-9\.]*\).*\( Build [0-9]*\)/\1\2\3/'`
+        ;;
+
+    *clang*)
+        cc_version_info="`$CC $CFLAGS $H4_CFLAGS --version 2>&1 |\
+            grep 'clang version' | sed 's/.*clang version \([-a-z0-9\.]*\).*/\1/'`"
+        ;;
+
+    *)
+        echo "No match to get cc_version_info for $CC"
+        ;;
 esac
 
-case $F77_BASENAME in 
-  g77)
-    FFLAGS="$FFLAGS"
-    DEBUG_FFLAGS="-g"
-    PROD_FFLAGS="-O"
-    PROFILE_FFLAGS="-pg"
-    ;;
-
-  *)
-    FFLAGS="$FFLAGS"
-    DEBUG_FFLAGS="-g"
-    PROD_FFLAGS="-O"
-    PROFILE_FFLAGS="-pg"
-    ;;
+fi
+
+# get fortran version info
+case $F77 in
+    *gfortran*)
+        fc_version_info=`$F77 $FFLAGS --version 2>&1 |\
+            grep 'GCC' | sed 's/\(.*(GCC) [-a-z0-9\. ]*\).*/\1/'`
+        fc_version="`$F77 $FFLAGS -v 2>&1 |grep 'gcc version' |\
+                sed 's/.*gcc version \([-a-z0-9\.]*\).*/\1/'`"
+        if test X != "X$fc_version"; then
+            fc_version=`echo $fc_version |sed 's/[-a-z]//g'`
+        fi
+        ;;
+
+    *ifc*|*ifort*)
+        fc_version_info=`$F77 $FFLAGS -V 2>&1 | grep 'Version' |\
+            sed 's/\(Intel.* Compiler\).*\( Version [a-z0-9\.]*\).*\( Build [0-9]*\)/\1\2\3/'`
+        fc_version="`$F77 $FFLAGS -V 2>&1 |grep '^Intel'`"
+        if test X != "X$fc_version"; then
+            fc_version="`echo $fc_version |sed 's/.*Version \([-a-z0-9\.\-]*\).*/\1/'`"
+        fi
+        ;;
+
+    *f95*)
+        # Figure out which compiler we are using: pgf90 or Absoft f95
```

Comment:
I don't see pgf90

Reply:
It's copied over verbatim from HDF5. Since I plan to bring over PGI support in a day or three, I figured I'd leave it in.